src/AzureAppConfigurationImpl.ts

```diff
@@ -60,25 +101,111 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
             };
             if (this.#requestTracingEnabled) {
                 listOptions.requestOptions = {
-                    customHeaders: this.#customHeaders()
+                    customHeaders: this.#customHeaders(requestType)
                 }
             }
 
             const settings = this.#client.listConfigurationSettings(listOptions);
 
             for await (const setting of settings) {
                 if (setting.key) {
-                    const [key, value] = await this.#processAdapters(setting);
-                    const trimmedKey = this.#keyWithPrefixesTrimmed(key);
-                    keyValues.push([trimmedKey, value]);
+                    const keyValuePair = await this.#processKeyValues(setting);
+                    keyValues.push(keyValuePair);
+                }
+                // update etag of sentinels if refresh is enabled
+                if (this.#refreshEnabled) {
+                    const matchedSentinel = this.#sentinels.find(s => s.key === setting.key && s.label === setting.label);
+                    if (matchedSentinel) {
+                        matchedSentinel.etag = setting.etag;
+                    }
                 }
             }
         }
+        this.clear(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.set(k, v);
         }
     }
 
+    /**
+     * Refresh the configuration store.
+     */
+    public async refresh(): Promise<void> {
+        if (!this.#refreshEnabled) {
+            return Promise.resolve();

```

Comment:
If someone tries to invoke `refresh()` without setting `refreshOptions.enabled=true`, should we throw an error instead of silently retuning?

Reply:
It's a good question. Per last design and the refresh behavior [above](https://github.com/Azure/AppConfiguration-JavaScriptProvider/pull/21#issue-1962913478), it's "no refresh", meaning we don't support refresh for the case.

I agree with you to explicitly throw an error.

Reply:
updated in 506c8a6